sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BatchScanExec.scala

```diff
@@ -241,3 +243,24 @@ case class BatchScanExec(
     s"BatchScan ${table.name()}".trim
   }
 }
+
+case class StoragePartitionJoinParams(
+    keyGroupedPartitioning: Option[Seq[Expression]] = None,
+    commonPartitionValues: Option[Seq[(InternalRow, Int)]] = None,
+    applyPartialClustering: Boolean = false,
+    replicatePartitions: Boolean = false) {
+  override def equals(other: Any): Boolean = other match {
```

Comment:
This is not new in this PR, but it's great if we can add some comments about why we exclude `keyGroupedPartitioning` in the comparison. cc @sunchao

Reply:
sure will do @cloud-fan (sorry just saw this!).